.github/dependabot.yml

```diff
@@ -0,0 +1,16 @@
+# Dependabot version updates raises a maximum of five pull requests each time it checks dependencies.
+# Note that there is some overlap with Dependabot security updates so some options can effect security updates as well,
+# see https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.
+
+version: 2
+updates:
+  # Maintain dependencies for npm ecosystem
+  - package-ecosystem: 'npm'
+    # Checks all directories from the current layer and below recursively for package.json files
+    directories:
+      - '**/*'
+    schedule:
+      # Runs every Monday
+      interval: 'weekly'
+    # Update package.json files if new version is outside of version range specified there. Otherwise lock file only.
+    versioning-strategy: increase-if-necessary
```

Comment:
If this strategy can update package.json files, how will we accommodate `changesets` in those PRs?

Reply:
That's a good point. Best if this could be automated.

If it can't:
PR won't pass checks until changeset is added (if our check works)
Maintainer adds changeset (this is doable using UI here https://github.com/aws-amplify/amplify-backend/pull/2353#issuecomment-2552462457 +/- quotes style but we could exclude changeset files from prettier.)

Reply:
Good point on `changesets`, I'll see if I can automate it.

Reply:
With change in [d826612](https://github.com/aws-amplify/amplify-backend/pull/2353/commits/d826612363cdea40813b88cb42908c1a957e58e8), I used the new script locally for this PR https://github.com/rtpascual/amplify-backend-fork/pull/20 and this is the resulting changeset file that will be force pushed to the PR:
<img width="535" alt="Screenshot 2024-12-19 at 5 18 28 PM" src="https://github.com/user-attachments/assets/4056a377-6213-41a5-b6c0-0bed733290f7" />